src/main/java/array/Exer09.java

```diff
@@ -0,0 +1,20 @@
+package array;
+
+public class Exer09 {
+
+    public int binarySearch(int[] array, int n) {
+        int start = 0;
+        int end = array.length;
+
+        while (end - start > 1) {
+            int middle = (start + end) >>> 1;
+            if (array[middle] == n) return middle;
+            if (n > array[middle]) {
+                start = middle;
+            } else {
+                end = middle;
+            }
+        }
+        return array[0] == n ? 0 : -1;
```

Comment:
Dòng này chứng tỏ em vẫn chưa hiểu về thuật toán

Reply:
Dạ cái này là em check xem mảng này có 1 phần tử hay không ạ.
Nếu lớn hơn 1 phần tử thì chỗ này là return -1 ạ

Reply:
Thế thì sai quy tắc rồi. Chưa kể chổ này mà mảng không có phần tử nào thì nó lỗi ngay lúc gọi `array[0]`

Reply:
Dạ.
Em mới sửa lại chút, anh kiểm tra giúp em với ạ :)